api/prometheus/v1/api_test.go

```diff
@@ -849,15 +867,14 @@ func TestAPIClientDo(t *testing.T) {
 				if err == nil {
 					t.Fatalf("expected error %q but got none", test.expectedErr)
 				}
-				if test.expectedErr.Error() != err.Error() {
-					t.Errorf("unexpected error: want %q, got %q", test.expectedErr, err)
-				}
+				testutil.Equals(t, test.expectedErr.Error(), err.Error(), "unexpected error: want %q, got %q", test.expectedErr, err)
+
 				if test.expectedErr.Detail != "" {
 					apiErr := err.(*Error)
-					if apiErr.Detail != test.expectedErr.Detail {
-						t.Errorf("unexpected error details: want %q, got %q", test.expectedErr.Detail, apiErr.Detail)
-					}
+					testutil.Equals(t, apiErr.Detail, test.expectedErr.Detail, "unexpected error details: want %q, got %q", test.expectedErr.Detail, apiErr.Detail)
```

Comment:
```suggestion
					testutil.Equals(t, apiErr.Detail, test.expectedErr.Detail)
```

no need of a message it is more less the default message from `testutil.Equals` anyway